src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionAbbreviationAsWordInNameTest.java

```diff
@@ -39,7 +39,7 @@ protected String getCheckName() {
 
     @Test
     public void testAnnotation() throws Exception {
-        final File fileToProcess = new File(getPath(
+        final File fileToProcess = new File(getXpathPath(
```

Comment:
This a bit weird, test class takes files from two locations: compilable and no compilable.
So no matter what type of test it is (xpath or it or regular) we just need getPath method.
Common base class can override it one for all.

Reply:
We would have change the current methods to be non-final, meaning any class can override them. If thats fine, then I will make the change.

Reply:
> We would have change the current methods to be non-final...

I would prefer to have another level of abstraction here where we set the path for each type of test(regular, xpath, ...) and then allow the respective tests to extend these new classes only. @rnveach what do you think about this?

Reply:
I lost a reason of my question. I am recalling it.

@nrmancuso, please recheck you point also. let @rnveach know if this is still issue for you.